flink-table-store-connector/src/main/java/org/apache/flink/table/store/connector/TableStoreFactory.java

```diff
@@ -129,6 +155,85 @@ public void onDropTable(Context context, boolean ignoreIfNotExists) {
         throw new UnsupportedOperationException("Not implement yet");
     }
 
+    @Override
+    public DynamicTableSource createDynamicTableSource(Context context) {
+        boolean streaming =
+                context.getConfiguration().get(ExecutionOptions.RUNTIME_MODE)
+                        == RuntimeExecutionMode.STREAMING;
+        LogSourceProvider logSourceProvider = null;
+        if (enableChangeTracking(context.getCatalogTable().getOptions())) {
+            logSourceProvider =
+                    createLogStoreTableFactory()
+                            .createSourceProvider(
+                                    createLogStoreContext(context),
+                                    new LogStoreTableFactory.SourceContext() {
+                                        @Override
+                                        public <T> TypeInformation<T> createTypeInformation(
+                                                DataType producedDataType) {
+                                            return createTypeInformation(
```

Comment:
We can just wrap `Context context`.
Use `context. createTypeInformation `.
Ditto.